rust/nvtcache/tests/nvtcache_tests.rs

```diff
@@ -0,0 +1,131 @@
+use ::nvtcache::dberror::Result;
+use ::nvtcache::nvt::Nvt;
+use ::nvtcache::redisconnector::KbNvtPos;
+use nvtcache::nvtcache;
+
+//test
+#[cfg(test)]
+mod test {
+    use std::env;
+
+    use ::nvtcache::nvt::NvtPref;
+
+    use super::*;
+
+    #[test]
+    #[cfg(feature = "redis_test")]
+    fn test_nvtcache() -> Result<()> {
+        let mut nvtcache: nvtcache::NvtCache;
+
+        let redis_default_socket = |_| "unix:///run/redis/redis-server.sock".to_string();
+        let redis_socket = env::var("REDIS_SOCKET").unwrap_or_else(redis_default_socket);
+        let default_plugin_path = |_| "/var/lib/openvas/plugins/".to_string();
+        let plugin_path = env::var("PLUGIN_PATH").unwrap_or_else(default_plugin_path);
+        let n = nvtcache::NvtCache::init(&redis_socket, &plugin_path);
+        match n {
+            Ok(nc) => nvtcache = nc,
+            Err(e) => {
+                println!("{}", e);
+                panic!("Error")
+            }
+        }
+
+        if nvtcache.is_init() {
+            println!("Is initialized");
+        }
+
+        match nvtcache.cache.get_namespace() {
+            Ok(ok) => println!("The namespace: {}", ok),
```

Comment:
Instead of println! you can use assert_eq to verify on ok and use the ? operation to return an Error when it is an error